### PR TITLE
feat: support templateParameters function

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/webpack-builder/src/plugins/html.ts [hunk 1]
```diff
@@ -96,6 +97,9 @@ async function getTemplateParameters(
       options: pluginOptions,
     },
     ...baseParameters,
+    ...(typeof templateParams === 'function'
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/html.ts:100]: 感觉可以考虑用 applyOptionsChain 来实现，把默认的 templateParameters 都传进去

File: packages/builder/webpack-builder/src/plugins/html.ts [hunk 2]
```diff
@@ -71,32 +71,36 @@ async function getTemplateParameters(
...
+  const templateParams =
+    templateParametersByEntries?.[entryName] || templateParameters;
   const baseParameters = {
     meta,
     title,
     mountId: mountId || DEFAULT_MOUNT_ID,
     entryName,
     assetPrefix,
-    ...(templateParametersByEntries?.[entryName] || templateParameters),
+    ...(typeof templateParameters === 'function' ? {} : templateParams),
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/plugins/html.ts:88]: 这一行应该多余了喔，applyOptionsChain 会进行 Object.assign

File: packages/builder/webpack-builder/src/types/config/html.ts [hunk 3]
```diff
@@ -16,9 +16,13 @@ export interface HtmlConfig {
   mountId?: string;
   crossorigin?: boolean | CrossOrigin;
   disableHtmlFolder?: boolean;
-  templateParameters?: Record<string, unknown>;
+  templateParameters?:
+    | Record<string, unknown>
+    | HTMLPluginOptions['templateParameters'];
```
Comment (chenjiahan) [inline packages/builder/webpack-builder/src/types/config/html.ts:21]: 类型可以更新下哈，文档里也同步改下~

**Review by chenjiahan — DISMISSED** [review]

**Comment by sanyuan0704** [comment]
刚解了下冲突 cc @chenjiahan

**Review by chenjiahan — APPROVED** [review]

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**